crates/dbsp/src/operator/trace.rs

```diff
@@ -391,18 +438,51 @@ where
         &self,
         bounds_stream: &Stream<C, TS>,
         retain_key_func: RK,
+    ) -> Stream<C, Spine<<B as SpillableBatch>::Spilled>>
+    where
+        B: Batch<Time = ()> + SpillableBatch + Send,
+        TS: DBData,
+        RK: Fn(&B::Key, &TS) -> bool + Clone + 'static,
+    {
+        self.shard()
+            .spill()
+            .integrate_trace_retain_keys_in_memory(bounds_stream, retain_key_func)
```

Comment:
For fault tolerance I think we will want all traces checkpointed.

Reply:
> For fault tolerance I think we will want all traces checkpointed.

I think that's true.

@ryzhyk We talked about how we don't need to store traces that will remain small on disk. That's true for storage, I think, but not for fault tolerance, where we need all traces to be on disk. Does that sound correct?